### PR TITLE
{compiler}[foss/2017b] LLVM v5.0.0

### PR DESCRIPTION
**Pull request by Helios07**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-foss-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+]
+
+dependencies = [
+    ('ncurses', '6.0'),
+    ('zlib', '1.2.11'),
+]
+
+configopts = '-DBUILD_SHARED_LIBS=ON '
+# required to install extra tools in bin/
+configopts += '-DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_ZLIB=ON '
```
Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-foss-2017b.eb]: @Helios07 I had to include `-DLLVM_ENABLE_RTTI=1` for the Mesa build on top of this LLVM installation to work correctly.

Have you already tried building and installing the Mesa easyconfig included in #5414 on top of this, and did it actually work?

**Comment by Helios07** [comment]
The Mesa build with this compiler worked for me, but since it doesn't seem to hurt, I included the compiler switch.

File: easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-foss-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ specified code representation known as the LLVM intermediate representation
+ ("LLVM IR"). The LLVM Core libraries are well documented, and it is
+ particularly easy to invent your own language (or port an existing compiler)
+ to use LLVM as an optimizer and code generator."""
+
+toolchain = {'name': 'foss', 'version': '2017b'}
+toolchainopts = {'cstd': 'gnu++11'}
+
+source_urls = ["http://llvm.org/releases/%(version)s"]
+sources = ["llvm-%(version)s.src.tar.xz"]
```
Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-foss-2017b.eb]: @Helios07 please include a SHA256 checksum

File: easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-foss-2017b.eb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+]
+
+dependencies = [
+    ('ncurses', '6.0'),
+    ('zlib', '1.2.11'),
+]
+
+configopts = '-DBUILD_SHARED_LIBS=ON '
+# required to install extra tools in bin/
+configopts += '-DLLVM_ENABLE_RTTI=ON -DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_ZLIB=ON '
```
Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-foss-2017b.eb]: Do you mind adding `-DLLVM_ENABLE_RTTI=ON` at the end (like it is in existing LLVM 5.x easyconfigs)?

```python
configopts += "-DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_ZLIB=ON -DLLVM_ENABLE_RTTI=1 "

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2548.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/59028d5d45695980c157e83b403df2f4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2131.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b1d850e278dd727311a6b98823b82c32 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Helios07!

**Pull request merged**